apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -669,6 +671,15 @@ export class ApiModelGenerator {
     this._processChildDeclarations(astDeclaration, exportedName, apiInterface);
   }
 
+  private _determineReadonly(astDeclaration: AstDeclaration): boolean {
+    const apiItemMetadata: ApiItemMetadata = this._collector.fetchApiItemMetadata(astDeclaration);
+    const docComment: tsdoc.DocComment | undefined = apiItemMetadata.tsdocComment;
+    const declarationMetadata: DeclarationMetadata = this._collector.fetchDeclarationMetadata(astDeclaration);
+    return (astDeclaration.modifierFlags & ts.ModifierFlags.Readonly) !== 0
+      || (docComment !== undefined &&  docComment.modifierTagSet.hasTagName('@readonly'))
```

Comment:
Suppose someone puts a `@readonly` tag on an item that cannot be readonly (i.e. they use the tag incorrectly). For example:

```
class Example {
  /** @readonly */
  set foo(value: string) { throw new Error('Cannot set foo'); }
}
```

Suppose `set foo` has no corresponding getter. It's obviously not readonly, because there's only a setter. But it has the `@readonly` tag, which means the developer has indicated that it should be displayed as readonly. What should we do in this case?

The logic as we have it will display it as readonly. I think that's _probably_ fine... to defer to the developer's understanding and just let the doc comment tag win. Just thought I'd call out this edge case.

Reply:
I'd think the inclusion of the comment that is specifically for documentation would over-ride poor implementation, especially in the context of a documentation generation tool, but in reality the comment has no meaning.

My 2c is that we would still evaluate the comment as isReadonly.